### PR TITLE
LPD-20239 add unit tests for doAsUserId bug

### PR DESCRIPTION
**Pull request by marcogalluzzi**
https://liferay.atlassian.net/browse/LPD-20239

This PR should be `on hold` until PR #5251 is merged into master

Commit  e918c11 has been added after commit 8cee0a527b3b6847cddd8eea8bd9b0772ca2a815 by Brian

### CODE REVIEW

**Comment by marcogalluzzi** [comment]
ci:test:sf

File: modules/apps/document-library/document-library-web/src/test/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilderTest.java [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+		mockHttpServletRequest.setAttribute(
+			WebKeys.THEME_DISPLAY, themeDisplay);
+
+		return new UIItemsBuilder(
+			mockHttpServletRequest, _fileEntry, _fileVersion, null, null,
+			_dlurlHelper);
+	}
+
+	private final DLURLHelper _dlurlHelper = Mockito.mock(DLURLHelper.class);
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/test/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilderTest.java]: Done. See new commit.

File: modules/apps/document-library/document-library-web/src/test/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilderTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		LanguageUtil languageUtil = new LanguageUtil();
+		PortalUtil portalUtil = new PortalUtil();
+
+		languageUtil.setLanguage(Mockito.mock(Language.class));
+		portalUtil.setPortal(new PortalImpl());
+	}
+
+	@Before
+	public void setUp() {
+		Mockito.when(
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/test/java/com/liferay/document/library/web/internal/display/context/logic/UIItemsBuilderTest.java:50]: Done. See new commit.

File: modules/apps/document-library/document-library-web/src/test/java/com/liferay/document/library/web/internal/display/context/DLViewEntriesDisplayContextTest.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+		return new DLViewEntriesDisplayContext(
+			new MockLiferayPortletRenderRequest(mockHttpServletRequest),
+			new MockLiferayPortletRenderResponse());
+	}
+
+	private static MockedStatic<DLURLHelperUtil> _dlURLHelperUtilMockedStatic;
+	private static MockedStatic<FrameworkUtil> _frameworkUtilMockedStatic;
+
+	private final FileVersion _fileVersion = Mockito.mock(FileVersion.class);
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/document-library/document-library-web/src/test/java/com/liferay/document/library/web/internal/display/context/DLViewEntriesDisplayContextTest.java]: Done. See new commit.

**Review by Mikellorza — APPROVED** [review]

**Comment by Mikellorza** [comment]
ci:forward

**Comment by marcogalluzzi** [comment]
ci:report:2553818940

**Comment by marcogalluzzi** [comment]
ci:forward:force

**Pull request closed without merging**